peps/pep-0705.rst

```diff
@@ -563,88 +373,169 @@ Suggestion for changes to the :mod:`typing` module, in line with current practic
 * Add ``typing.ReadOnly``, linked to TypedDict and this PEP.
 * Add the following text to the TypedDict entry:
 
-By default, keys not specified in a TypedDict may still be present. Instances can be restricted to only the named keys with the ``other_keys`` flag. *insert example, perhaps using ``in`` to illustrate the benefit*
-
-Individual keys can be excluded from mutate operations using ReadOnly, allowing them to be read but not changed. This is useful when the exact type of the value is not known yet, and so modifying it would break structural subtypes. *insert example*
+Individual items can be excluded from mutate operations using ReadOnly, allowing them to be read but not changed. This is useful when the exact type of the value is not known yet, and so modifying it would break structural subtypes. *insert example*
 
-If all keys on a TypedDict should be read-only, the ``readonly`` flag can be used as a shorthand. *insert example*
-
-Reference Implementation
+Reference implementation
 ========================
 
 No complete reference implementation exists yet. pyright 1.1.310 ships with a partial implementation of the ReadOnly qualifier.
```

Comment:
The current version of pyright (1.1.332) fully implements the current proposal. In the process of manual testing with your samples, I found and fixed a few bugs that will be included in this week's version (1.1.333).